analogout.py

```diff
@@ -148,35 +159,51 @@ def init(self):
-                if self.externalClock.useExternalClock:
-                    self.task.timing.cfg_samp_clk_timing(
-                        rate=self.externalClock.maxClockRate, 
-                        source=self.externalClock.source, 
-                        active_edge=Edge.RISING, # default
-                        sample_mode=AcquisitionType.FINITE, # default
-                        samps_per_chan=1000) # default
-                    
-                if self.startTrigger.wait_for_start_trigger:
-                    self.task.start_trigger.cfg_dig_edge_start_trig(
-                        trigger_source=self.startTrigger.source,
-                        trigger_edge=self.startTrigger.edge) # default
-                                    
-                if self.exportStartTrigger:
-                    self.task.export_signals.export_signal(
-                        Signal.START_TRIGGER,
-                        self.exportTrigger.outputTerminal)
+                try:
+                    if self.task is not None:
+                        try:
+                            self.task.close()
+                            
+                        except DaqError as e:
+                            # end the task nicely
+                            self.stop()
+                            self.close()
+                            msg = '\n AnalogOutput failed to close current task'
+                            raise HardwareError(self, task=self.task, message=msg)
+                        
+                try:
+                    self.task = nidaqmx.Task() # might be task.Task()
+                    self.task.ao_channels.add_ao_voltage_chan(
+                        self.physicalChannels,
+                        min_val = self.minValue,
+                        max_val = self.maxValue)
                     
-                self.isInitialized = True
+                    if self.externalClock.useExternalClock:
+                        self.task.timing.cfg_samp_clk_timing(
+                            rate=self.externalClock.maxClockRate, 
+                            source=self.externalClock.source, 
+                            active_edge=Edge.RISING, # default
+                            sample_mode=AcquisitionType.FINITE, # default
+                            samps_per_chan=1000) # default
+                        
+                    if self.startTrigger.wait_for_start_trigger:
+                        self.task.start_trigger.cfg_dig_edge_start_trig(
+                            trigger_source=self.startTrigger.source,
+                            trigger_edge=self.startTrigger.edge) # default
+                                        
+                    if self.exportStartTrigger:
+                        self.task.export_signals.export_signal(
+                            Signal.START_TRIGGER,
+                            self.exportTrigger.outputTerminal)
+                
+                except DaqError as e:
```

Comment:
same as previous. Won't mark every case like I did with self.task to reduce spam this time though.